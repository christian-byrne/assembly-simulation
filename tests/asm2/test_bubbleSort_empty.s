.data

COMMA_SEP:
	.asciiz ", "
    
.globl   printInts
printInts:
	.byte 1

.globl   printWords
printWords:
	.byte 0

.globl   bubbleSort
bubbleSort:
	.byte 1

.globl   printInts_howToFindLen
printInts_howToFindLen:
	.half 1

.globl   intsArray
intsArray:
	

.globl   intsArray_END
intsArray_END:
	

.globl   intsArray_len
intsArray_len:
	.word 0

.globl   theString
theString:
	.asciiz "DO NOT TOUCH!"


.globl   PVqeHesPsFNw
PVqeHesPsFNw:
	.asciiz "\n+-------------------------------+\n|    Testcase Variable Dump:    |\n+-------------------------------+\n\n"

.globl   wqkckowDXtMn
wqkckowDXtMn:
	.asciiz "==============================================\nTEST (bubbleSort_empty.s):\n\nprintInts=1\nprintWords=0\nbubbleSort=1\nprintInts_howToFindLen=1\nintsArray=[]\nintsArray_len=0\ntheString=\"DO NOT TOUCH!\"\n\n==============================================\n\n"


.text

.globl main

main:
	addi $v0, $zero, 4	# print
	la $a0, wqkckowDXtMn
	syscall

	jal studentMain

.text
	addi $v0, $zero, 4	# print
	la $a0, PVqeHesPsFNw
	syscall

	addi $v0, $zero, 11	# print
	addi $a0, $zero, '['
	syscall

	la $t0, intsArray
    
	

	add $t0, $zero, $zero   # reset t0
    
	addi $v0, $zero, 11	# print
	addi $a0, $zero, ']'
	syscall
	addi $v0, $zero, 11
	addi $a0, $zero, '\n'
	syscall

	addi $v0, $zero, 4	# print
	la $a0, theString
	syscall

	addi $v0, $zero, 11
	addi $a0, $zero, '\n'
	syscall


	addi $v0, $zero, 10
	syscall
