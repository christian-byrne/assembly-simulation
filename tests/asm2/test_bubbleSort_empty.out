==============================================
TEST (bubbleSort_empty.s):

printInts=1
printWords=0
bubbleSort=1
printInts_howToFindLen=1
intsArray=[]
intsArray_len=0
theString="DO NOT TOUCH!"

==============================================

printInts: About to print 0 elements.

+-------------------------------+
|    Testcase Variable Dump:    |
+-------------------------------+

[]
DO NOT TOUCH!
